Add Portkey as an OpenAI-compatible provider

vrushankportkey opened the pull request:
### Related Issue
**Issue:** New Issue

### Description
- Addressed @arafatkatze feedback by removing any Portkey‑specific implementation and introducing a generic, UI‑only preset for OpenAI‑compatible vendors.
  - Forward‑compatible: adding future OpenAI‑compatible vendors is a single preset entry; no backend or provider type changes needed.
- Model ID UX improved:
  - Fetched models auto-populate a dropdown; includes “Add manually…” to switch to manual input and “Pick from list” to return.
  - Auto‑selects the first fetched model if none is set.
- No proto/backend changes; reuses existing OpenAI‑compatible flow and `refreshOpenAiModels`.

Changed files:
- Updated: `webview-ui/src/components/settings/ApiOptions.tsx`
- Updated: `webview-ui/src/components/settings/providers/OpenAICompatible.tsx`
- New: `webview-ui/src/components/settings/utils/providerPresets.ts`

### Test Procedure
- In Settings → API Provider:
  - Select “Portkey”.
  - If `openAiBaseUrl` was empty, it should default to `https://api.portkey.ai/v1`.
  - Enter a valid Portkey API Key ([get from here](https://app.portkey.ai/)); pick a Model ID (auto‑populates via `/models`).
  - Optionally choose “Add manually…” to enter a custom model and “Pick from list” to return to the dropdown.

### Type of Change
-   [x] ✨ New feature (non-breaking change which adds functionality)
-   [ ] 🐛 Bug fix
-   [ ] 💥 Breaking change
-   [ ] ♻️ Refactor
-   [ ] 💅 Cosmetic
-   [ ] 📚 Documentation
-   [ ] 🏃 Workflow

### Pre-flight Checklist
-   [x] Changes are focused on a single feature
-   [x] Tests/lint/format pass locally (`npm test`, `npm run format && npm run lint`)
-   [x] Changeset created (`npm run changeset`)
-   [x] Reviewed CONTRIBUTING.md

### Screenshots
<img width="758" height="938" alt="CleanShot 2025-08-22 at 18 21 47@2x" src="https://github.com/user-attachments/assets/d8a22451-a8d3-4980-82ae-86de37c587d4" />

<!-- ELLIPSIS_HIDDEN -->


----

> [!IMPORTANT]
> Adds Portkey as an OpenAI-compatible provider with UI enhancements for model selection and documentation updates.
> 
>   - **Behavior**:
>     - Adds Portkey as an OpenAI-compatible provider with default `openAiBaseUrl` set to `https://api.portkey.ai/v1`.
>     - Enhances model ID selection UX in `OpenAICompatibleProvider.tsx` with auto-populated dropdown and manual entry toggle.
>   - **UI Components**:
>     - Updates `ApiOptions.tsx` to include Portkey in the provider dropdown.
>     - Introduces `providerPresets.ts` for managing OpenAI-compatible provider presets.
>   - **Documentation**:
>     - Adds `portkey.mdx` to document Portkey configuration and usage.
>     - Updates `docs.json` to include Portkey in the provider configuration section.
> 
> <sup>This description was created by </sup>[<img alt="Ellipsis" src="https://img.shields.io/badge/Ellipsis-blue?color=175173">](https://www.ellipsis.dev?ref=cline%2Fcline&utm_source=github&utm_medium=referral)<sup> for 7c7ffa9c057b3c9dcd4409efce68430a88d61b8f. You can [customize](https://app.ellipsis.dev/cline/settings/summaries) this summary. It will automatically update as commits are pushed.</sup>


<!-- ELLIPSIS_HIDDEN -->

vrushankportkey commented:
Hi @arafatkatze updated the PR based on your feedback. What do you think?

arafatkatze commented:
@vrushankportkey Sorry for some recent pushes being made to main if you can rebase this against the latest main I will test this again and merge.

vrushankportkey commented:
Hey @arafatkatze done now! Please check?

arafatkatze commented:
@vrushankportkey Would it be possible to just use portkey directly as something that's openAI compatible rather than adding a separate preset? 

WE can still have a documentation  page for it?

vrushankportkey commented:
Hi @arafatkatze thanks a lot for catching that! It is fixed now. I also made some more enhancements and tried to reduce the surface area of the PR. Removed the providerPresets file for now and just added that to the OpenAICompatible file directly. This is working beautifully now, and thank you for understanding our use case!


https://github.com/user-attachments/assets/c2b5cabd-0d5e-4bec-9de3-a03a4cdb87ae

arafatkatze commented:
Thanks a million for the PR btw, this is great.

arafatkatze commented on webview-ui/tsconfig.app.json:
```diff
@@ -50,5 +50,9 @@
 	"include": [
 		"src"
+	],
+	"exclude": [
```
@vrushankportkey Any reasons for this?

ClineXDiego commented:
This PR is unlikely to be merged, so we’re going to close it to keep the queue clean.

If you believe the underlying issue still isn’t resolved, please search for an existing issue (to avoid duplicates) or open a new one with repro details, then request assignment so we can track and review it first.

If this is a feature request, please open a Discussion first so the maintainers/community can weigh in before any implementation work.

Once a maintainer assigns/approves it, feel free to open a new PR and tag us here for review.
